### PR TITLE
docs: clarify "getting started" section

### PR DESCRIPTION
**Pull request by lpawelcz**
This PR renames the `Getting started` section to clarify that it is a part of toolchain description. It also emphasis this in the note and points user directly to user documentation instead of `symbiflow-examples` repository.

### CODE REVIEW

File: docs/source/getting-started.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
-===============
-Getting Started
-===============
+====================================================
+Getting Started with SymbiFlow Toolchain development
+====================================================
 
 .. note::
```
Comment (mithro) [inline docs/source/getting-started.rst:5]: Might be worth changing this to a warning?

**Review by mithro — APPROVED** [review]
One minor comment, otherwise LGTM!

File: docs/source/getting-started.rst (hunk 1 above)
Comment (lpawelcz) [reply docs/source/getting-started.rst]: That's a good idea - updated

**Review by mithro — APPROVED** [review]
LGTM!

**Pull request merged**